consul/watcher.go

```diff
@@ -454,9 +454,12 @@ func (w *Watcher) genCfg() Config {
 		}
 		for _, s := range up.Nodes {
 			serviceInstancesTotal++
-			host := s.Service.Address
-			if host == "" {
-				host = s.Node.Address
+
+			name := s.Node.Node
```

Comment:
name would not be unique: the proxy might try to target several instances of the same service on 1 single node